extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.js

```diff
@@ -45,14 +144,33 @@ require(
 				return $.proxy(func, this);
 			},
 
-			getBaseUrl: function() {
+			getBaseUrl: function () {
 				return mw.config.get('wgOnSiteNotificationsApiUrl');
 			}
 
 		};
 
+		function compileMustache() {
+			loader({
+				type: loader.MULTI,
+				resources: {
+					mustache: 'extensions/wikia/DesignSystem/services/templates/DesignSystemGlobalNavigationOnSiteNotifications.mustache',
+					scripts: 'design_system_on_site_notifications_mustache_js'
+				}
+			}).done(function (assets) {
+				loader.processScript(assets.scripts);
+				OnSiteNotifications.init(assets.mustache[0]);
+			});
+		}
+
+		function getSafely(obj, path) {
+			return path.split(".").reduce(function (acc, key) {
+				return (typeof acc == "undefined" || acc === null) ? acc : acc[key];
```

Comment:
This is far from a perfect solution but it works for chains where at some point the object may be either undefined or null.
The last object in the chain does not have to be an object, so ignoring everything but objects will fail the last step.